src/Polysemy/Scoped.hs

```diff
@@ -2,21 +2,86 @@ module Polysemy.Scoped (
   -- * Effect
   Scoped,
 
-  -- * Constructor
+  -- * Constructors
   scoped,
+  scoped_,
+  rescope,
 
   -- * Interpreters
-  runScoped,
-  runScopedAs,
   interpretScopedH,
+  interpretScopedH',
   interpretScoped,
   interpretScopedAs,
+  runScoped,
+  runScopedAs,
 ) where
 
 import Polysemy.Internal (InterpreterFor, Sem, liftSem, raise)
-import Polysemy.Internal.Scoped (Scoped (Run, InScope), scoped, interpretH')
-import Polysemy.Internal.Union (Weaving(Weaving), injWeaving)
-import Polysemy.Internal.Tactics (Tactical, runTactics)
+import Polysemy.Internal.Combinators (interpretH)
+import Polysemy.Internal.Scoped (Scoped (InScope, Run), interpretWeaving, rescope, scoped, scoped_)
+import Polysemy.Internal.Tactics (Tactical, runTactics, runTSimple, liftT)
+import Polysemy.Internal.Union (Weaving (Weaving), injWeaving)
+
+-- |Interpreter for 'Scoped', taking a @resource@ allocation function and a parameterized handler for the plain
+-- @effect@.
+--
+-- >>> runScoped withResource scopedHandler
+--
+-- @withResource@ is a callback function, allowing the user to acquire the resource for each program wrapped by 'scoped'
+-- using other effects, with an additional argument that contains the call site parameter passed to 'scoped'.
```

Comment:
Examples would be awesome if you can spare the time :sweat_smile: But I guess this is true for almost everything in the library, so it's not really very important to start adding them now.

Reply:
I added one to `Scoped`, and there is also the link to my blog post.
do you think we should reference the `Scoped` docs in each interpreter combi? instead of having any prose there?

Reply:
Awesome.

Yeah I think it makes sense to point to `Scoped` from all the combinators, although I expect anyone looking at the combinators to also be looking at `Scoped` (unless you're looking at the smart constructor `scoped`, in which case it might be still necessary)

Come to think of it, I think I thought of asking about an example here is because you wrote more docs here in the first place, which I thought would be better suited to have at the `Scoped` type. But I think right now it's fine as is. 

So feel free to resolve if you think it's not necessary to add links to `Scoped` from all the combinators

Reply:
> Come to think of it, I think I thought of asking about an example here is because you wrote more docs here in the first place, which I thought would be better suited to have at the Scoped type.

That's what I assumed! It's a reasonable concern though, I always struggle to decide whether to duplicate docs for names that have almost identical purposes.

In the HLS workflow, you would maybe start typing a common prefix and then cycle through the completions, reading the docs for each of them to decide which one is appropriate.
In that case, referring to a name that isn't listed due to no common prefix (`scoped` vs `Scoped`), you would have to change to a different workflow.
On the other hand, if you already _know_ that you want `interpretScopedX`, then you probably already know the fundamentals of the concept, and the information you need is how the specific `X` differs from the others, so only describing the specifics of the interpreter variant would make most sense.

In the workflow in which you're browsing Hackage to learn how it works, the reference is easily followed, and duplication of the prose would probably only cause cognitive overhead for finding the information that you haven't already read on the previous Haddock.

What do you think?

Reply:
I agree with your assessment.